src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -18468,6 +18479,509 @@ public void Main()
+                //         new COpen<T>().P1.ToString(); // 2
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "new COpen<T>().P1").WithLocation(10, 9),
+                // (11,9): warning CS8602: Dereference of a possibly null reference.
+                //         new CClass<T>().P2.ToString(); // 3
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "new CClass<T>().P2").WithLocation(11, 9),
+                // (12,9): warning CS8602: Dereference of a possibly null reference.
+                //         new CClass<T>().P3.ToString(); // 4
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "new CClass<T>().P3").WithLocation(12, 9),
+                // (18,9): warning CS8629: Nullable value type may be null.
+                //         new COpen<T?>().P1.Value.ToString(); // 5
+                Diagnostic(ErrorCode.WRN_NullableValueTypeMayBeNull, "new COpen<T?>().P1").WithLocation(18, 9),
+                // (19,9): warning CS8629: Nullable value type may be null.
+                //         new CStruct<T>().P5.Value.ToString(); // 6
+                Diagnostic(ErrorCode.WRN_NullableValueTypeMayBeNull, "new CStruct<T>().P5").WithLocation(19, 9)
+            };
+
+            var comp = CreateNullableCompilation(source, references: new[] { lib.EmitToImageReference() });
+            comp.VerifyDiagnostics(expected);
+
+            var comp2 = CreateNullableCompilation(new[] { source, lib_cs, MaybeNullAttributeDefinition });
+            comp2.VerifyDiagnostics(expected);
+        }
+
+        [Fact]
+        public void MaybeNull_Property_OnVirtualProperty()
+        {
+            // Consider warning on improper nullability attributes in OHI? https://github.com/dotnet/roslyn/issues/36039
+            var source = @"
+using System.Diagnostics.CodeAnalysis;
+public class Base
+{
+    [MaybeNull] public virtual string P => throw null!;
+    public virtual string P2 => throw null!;
+}
+public class C : Base
+{
+    public override string P => throw null!;
+    [MaybeNull] public override string P2 => throw null!;
+
+    static void M(C c, Base b)
+    {
+        b.P.ToString(); // 1
+        c.P.ToString();
+
+        b.P2.ToString();
+        c.P2.ToString(); // 2
+    }
+}";
+            var comp = CreateNullableCompilation(new[] { source, MaybeNullAttributeDefinition });
+            comp.VerifyDiagnostics(
+                // (15,9): warning CS8602: Dereference of a possibly null reference.
+                //         b.P.ToString(); // 1
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "b.P").WithLocation(15, 9),
+                // (19,9): warning CS8602: Dereference of a possibly null reference.
+                //         c.P2.ToString(); // 2
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "c.P2").WithLocation(19, 9)
+                );
+        }
+
+        [Fact]
+        public void MaybeNull_Property_OnVirtualProperty_OnlyOverridingSetter()
```

Comment:
>MaybeNull_Property_OnVirtualProperty_OnlyOverridingSetter [](start = 20, length = 57)

Do we have similar tests for attributes on setters?

Reply:
Added `Disallow_Property_OnVirtualProperty_OnlyOverridingGetter` and `Disallow_Indexer_OnVirtualIndexer_OnlyOverridingSetter` in iteration 7. Is that what you meant?

---
In reply to: [295923955](https://github.com/dotnet/roslyn/pull/36444#discussion_r295923955) [](ancestors = 295923955)